cmd/grype-db/cli/commands/cache_backup.go

```diff
@@ -138,13 +141,25 @@ func archiveProvider(root string, name string, writer *tar.Writer) error {
 
 	return filepath.Walk(name,
 		func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-
 			if info.IsDir() {
 				return nil
 			}
+			if strings.HasSuffix(path, "metadata.json") && cfg.Results.ResultsOnly {
+				// Mark metadata stale
+				state, err := provider.ReadState(path)
+				if err != nil {
+					return err
+				}
+
+				state.Stale = true
+				err = provider.WriteState(path, state)
```

Comment:
I think this will write the metadata back to the existing path in the vunnel workspace. We only want to affect the `metadata.json` that is being written to the archive (the vunnel workspace shouldn't be written to here).